wagtail/admin/views/bulk_actions/delete.py

```diff
@@ -0,0 +1,65 @@
+from urllib.parse import urlencode
+
+from django.core.exceptions import PermissionDenied
+from django.db import transaction
+from django.shortcuts import get_list_or_404, redirect
+from django.template.response import TemplateResponse
+from django.urls import reverse
+from django.utils.translation import gettext as _
+
+from wagtail.admin import messages
+from wagtail.admin.views.pages.utils import get_valid_next_url_from_request
+from wagtail.core import hooks
+from wagtail.core.models import Page
+
+
+def delete(request, parent_page_id):
+    next_url = get_valid_next_url_from_request(request)
+    if not next_url:
+        next_url = reverse('wagtailadmin_explore', args=[parent_page_id])
+
+    page_ids = list(map(int, request.GET.getlist('id')))
+    pages = []
+
+    for page in get_list_or_404(Page, id__in=page_ids):
+        page = page.specific
+        if not page.permissions_for_user(request.user).can_delete():
+            raise PermissionDenied
+        pages.append(page)
+        for fn in hooks.get_hooks('before_delete_page'):
+            result = fn(request, page)
+            if hasattr(result, 'status_code'):
+                return result
+
+    if request.method == 'GET':
+        _pages = []
+        for page in pages:
+            _pages.append({
+                'page': page,
+                'descendant_count': page.get_descendant_count(),
+            })
+
+        return TemplateResponse(request, 'wagtailadmin/pages/bulk_actions/confirm_bulk_delete.html', {
+            'pages': _pages,
+            'next': next_url,
+            'submit_url': (
+                reverse('wagtailadmin_bulk_delete', args=[parent_page_id])
+                + '?' + urlencode([('id', page_id) for page_id in page_ids])
+            ),
+        })
+    elif request.method == 'POST':
+        num_parent_pages = 0
+        num_child_pages = 0
+        with transaction.atomic():
+            for page in pages:
+                num_parent_pages += 1
+                num_child_pages += page.get_descendant_count()
+                page.delete(user=request.user)
+
+                for fn in hooks.get_hooks('after_delete_page'):
+                    result = fn(request, page)
+                    if hasattr(result, 'status_code'):
+                        return result
+
+        messages.success(request, _(f'You have successfully deleted {num_parent_pages} pages including {num_child_pages} child pages.'))
```

Comment:
suggestion: use proper pluralisation - https://docs.djangoproject.com/en/3.2/topics/i18n/translation/#pluralization

The project convention is not to use f-strings where translation is needed. Instead use % formatting. More details in #6238

_("%(time_period)s ago") % {'time_period': time_period}

Additionally, the message could read something along the lines of `_("%(parent_pages)d pages and %(child_pages)d child pages" % {'parent_pages': num_parent_pages, 'child_pages': num_child_pages })` -- most action success messages are impersonal. ie. "Page 'TITLE' deleted. " or "Page 'TITLE' has been published." instead of `You have successfully deleted page "TITLE"`

I suggest changing for all actions